R/MsExperiment.R

```diff
@@ -214,48 +214,99 @@ setMethod("readMsObject", signature(object = "MsExperiment",
 setMethod("readMsObject",
           signature(object = "MsExperiment",
                     param = "MetaboLightsParam"),
-          function(object, param, ...) {
-              url <- file.path(
-                  "https://ftp.ebi.ac.uk/pub/databases/metabolights/studies/public/",
-                  param@studyId, "/")
-              ## Retrieve the HTML content using curl system command
-              html_content <- system(paste("curl -s", url), intern = TRUE) ##idk if that works for all OS
-              if (all(grepl("The requested URL was not found on this server",
-                       html_content)))
-                  stop("Study not found. Please check the study ID and try again.")
-              files <- regmatches(html_content,
-                                  gregexpr("href=\"([^\"]+\\.txt|[^\"]+\\.tsv)\"",
-                                           html_content))
-              files <- unlist(files)
-              files <- gsub("href=\"|\"", "", files)
+          function(object, param, keepOntology = TRUE, keepProtocol = TRUE,
+                   simplify = TRUE, ...) {
```

Comment:
maybe add a `check.names = FALSE` parameter too? that one could be passed to both the `read.table()` calls as well as the `object@sampleData <- DataFrame` call. To me that would be a nice parameter that allows users to configure how they want the sample data column names to be processed - and saves us from defining how to handle white spaces in column names etc.